src/server/api/v1/controllers/effects-api-controller.ts

```diff
@@ -96,7 +106,7 @@ async function runPresetEffectList(req, res, waitForCompletion = false) {
 
     const body = req.body || {};
     const query = req.query || {};
-    let args, username;
+    let args: ParsedQs, username: string | ParsedQs | string[] | ParsedQs[];
```

Comment:
This looks weird to me. 
Shouldn't `username` be a string ? 
As this is, it seems to me that we're just passing along something that could very well be a dictionary. i think there should be input validation instead here, and making sure username is only ever a string or throwing a malformed request error.

Reply:
thats the type of the query